core/tracepoint/enable_spec.rb

```diff
@@ -0,0 +1,50 @@
+require File.expand_path('../../../spec_helper', __FILE__)
+
+ruby_version_is '2.0' do
+  describe 'TracePoint#enable' do
+    def test; end
+
+    it 'returns true if trace was enabled' do
+      event_name, method_name = nil
+      trace = TracePoint.new(:call) do |tp|
+        event_name = tp.event
+        method_name = tp.method_id
+      end
+
+      event_name.should equal(nil)
+      test
+      method_name.equal?(:test).should be_false
+
+      trace.enable
+      trace.enable.should be_true
+      event_name.should equal(:call)
+      test
+      method_name.equal?(:test).should be_true
+    end
```

Comment:
trace is still enabled at this point

Reply:
Reply to both(`enable { 'with_block' }` & this comment ^) comments - 
How about this change for line 18-22 - 
```
trace.enable do
  trace.enable.should be_true
  event_name.should equal(:call)
  test
  method_name.equal?(:test).should be_true
end
``` 
OR should we add separate specs for `enable {'with_block'}` ?

Reply:
Definitely separate specs for `enable {'with_block'}`

Reply:
PS: You can check how other specs are written for methods that take a block or not, e.g. `File.open`

Reply:
I have added the specs for `enable {'with_block'}`
Can you please check?

Reply:
For the second spec, I'd recommend using a `lambda` to show that it doesn't pass any parameter; in Ruby, any block can accept extra parameters that will be filled with `nil`.

It's important to test that the trace point is not enabled after the call to `enable {}`. Also, what's the result of `enable{}`? Also, what happens if the trace point is already enabled and we call `enable {}`? Is it still enabled after?

Reply:
Or instead of a lambda, you could capture `|*args|` and see that `args` is empty

Reply:
okay, I have updated the specs.